src/index.js

```diff
@@ -2,40 +2,35 @@
 
 import simpleOAuth2 from 'simple-oauth2';
 import url from 'url';
-import BynderApi from './api';
+import ApiWrapper from './api';
 import {rejectValidation, bodyTypes, getLength, create256HexHash} from './utils';
 import {DEFAULT_ASSETS_NUMBER_PER_PAGE, FILE_CHUNK_SIZE, FORM_ENCODED_HEADER} from './constants';
 
-/**
- * @classdesc Represents the Bynder SDK. It allows the user to make every call to the API with a single function.
- * @class
- */
-export default class Bynder {
+/** Represents the Bynder SDK. It allows the user to make every call to the API with a single function */
+class Bynder {
   /**
    * Create Bynder SDK.
    * @constructor
+   * @param {Object} options An object containing the consumer keys, access keys and the base URL.
    * @param {String} options.baseURL The URL with the account domain
    * @param {String} options.clientId OAuth2 client id
    * @param {String} options.clientSecret OAuth2 client secret
    * @param {String} options.redirectUri Redirection URI
-   * @param {String} options.permanentToken Optional permanent token
    * @param {String} options.token.access_token Optional access token
    * @param {String} options.httpsAgent Optional https agent
    * @param {String} options.httpAgent Optional http agent
-   * @param
-   * @param {Object} options An object containing the consumer keys, access keys and the base URL.
+   * @param {Boolean} options.clientCredentials Determines if we should use client credentials
    */
   constructor({baseURL, redirectUri, clientId, clientSecret, ...options}) {
+    if (options.permanentToken) {
+      throw new Error('Permanent tokens are no longer supported. Please use OAuth 2 authorization code or client credentials');
+    }
+
     this.baseURL = baseURL;
     this.redirectUri = redirectUri;
     this.options = options;
-
-    this.api = new BynderApi(baseURL, options.httpsAgent, options.httpAgent);
-
-    if (typeof options.permanentToken === 'string') {
-      this.api.permanentToken = options.permanentToken;
-      return;
-    }
+    this._hasClientCredentials = options.clientCredentials;
```

Comment:
We don't strictly need this field and param, since we can also check if a redirectUri has been passed or not.

Reply:
Fixed.